### ISSUE TITLE
Create Dockerfiles for IVA, Tomcat, Daemon, OpenCGA CLI container

### ISSUE THREAD

**Issue by marrobi** [opencb/opencga#949]
https://github.com/wbari/opencga/tree/docker/opencga-docker 

- [x] IVA

marrobi/iva - needs work around configuration.

- [ ] Tomcat
marrobi/opencga-app - Needs testing.

GRPC moved to separate issue.

- [ ] Daemon
marrobi/opencga-daemon

- [ ] opencga-init

- [ ] OpenCGA CLI
marrobi/opencga
Container with OpenCGA, anything else?

**Comment by marrobi** [comment]
Need to know how to run daemon non-interactively.

**Comment by marrobi** [comment]
Four docker images succesfully build and run - opencga-java-base, opencga-build opencga, opencga-daemon, opencga-app. Further testing needed, but have used daemon to do a transform.

https://github.com/marrobi/opencga/tree/feature/loginvm/opencga-app/app/scripts/docker

**Comment by marrobi** [comment]
Cannot finish docker image startup scripts until have mongo to connect to. #952 or #951 . Could create temporary Mongo install if easier.

**Comment by imedina** [comment]
I hope this helps:
- IVA is a web app, it only needs Apache HTTPD or similar. If we want to be able to build the web app then we need nodejs and npm (and Git). This a hard requirement, it must be installed
- GRPC provides an alternative RPC for querying variants and BAM files. The main RPC is REST. This is not a hard requirement. It will be probably needed but right now.

**Comment by marrobi** [comment]
Trying to reduce container size/build time, but blocked on:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (validate) on project opencga: Execution validate of goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check failed: Plugin org.apache.maven.plugins:maven-checkstyle-plugin:2.17 or
one of its dependencies could not be resolved: The following artifacts could not be resolved: com.puppycrawl.tools:checkstyle:jar:8.1, commons-lang:commons-lang:jar:2.1: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact com.puppycrawl.tools:checkstyle:jar:8.1 has not been downloaded from it before. -> [Help 1]
```

**Comment by marrobi** [comment]
@imedina to provide info on GRPC
@j-coll to discuss offline dependency issue on Slack.

**Comment by jjcollinge** [comment]
Add NFS volume from init to app and daemon as readonly volume

**Comment by marrobi** [comment]
Closing as done in project.
